### DISCUSSION TITLE
Inverting tableaux to improve complexity of deterministic measurement

### DISCUSSION THREAD

**Discussion by Fe-r-oz** [QuantumSavory/QuantumClifford.jl#433] [Q&A] [answered]
Hi Stefan!

I’m pretty sure you've considered the functionality of inverting tableaux, as discussed in [2.3.3 Inverting Tableaus](https://arxiv.org/pdf/2103.02202). My question pertains to the following context: "stim improves the asymptotic complexity of deterministic measurement from quadratic to linear by efficiently tracking the inverse of the circuit's stabilizer tableau". On a small tangent: There are spelling errors in the aforementioned paper as Tableaus is used 👀 

Is it meaningful for us to implement inverting tableaux, or is there currently a workaround for this?

Thank you!

**Comment by Krastanov** [comment, marked as answer]
Inverting a Clifford operation is easy, with the `inv` function. What stim does is not really inverting a tableau after the fact, rather it just always tracks the application of the inverse Clifford operation. That would be nice to have and relatively easy to implement -- the difficulty might actually be just properly documenting and designing the API (and benchmarking to prove there is a performance improvement)

**Reply by Fe-r-oz** [reply]
Yup, it would be a nice good first issue as well!

Edit: #448

**Reply by Fe-r-oz** [reply]
Hi, Stefan!

Thank you for your positive reaction and considering my request! Please help improve the write up of the https://github.com/QuantumSavory/QuantumClifford.jl/issues/448 that describes this feature so that it can be turned into a good first issue 🙂  I tried to add your insights but your help would be greatly appreciated!
